client/pages/page-edit.jsx

```diff
@@ -61,19 +40,52 @@ var PageField = React.createClass({
   propTypes: {
     page: React.PropTypes.object,
     project: React.PropTypes.object,
-    field: React.PropTypes.string,
-    update: React.PropTypes.func,
-    save: React.PropTypes.func
+    field: React.PropTypes.string
   },
   getInitialState () {
     let page = this.props.page
     let field = this.props.field
     let type = (page.schema[field] && page.schema[field].type) || 'text'
     let content = page.content.json[field]
-    return { type, content }
+    let newContent = (content instanceof Object) ? Object.assign({}, content) : content
+    return { type, content, newContent }
```

Comment:
Yes, so it's informing the child component what to show - for example the existing colour in the case of the colour-picker.  `newContent` is modified by the `update` method (which is passed to the child component), and then read in the `save` method.  If we don't differentiate, we can't show existing and new colour.